azure/defaults.go

```diff
@@ -80,6 +80,11 @@ func GenerateFrontendIPConfigName(lbName string) string {
 	return fmt.Sprintf("%s-%s", lbName, "frontEnd")
 }
 
+// GenerateNatGatewayIPConfigName generates a nat gateway IP config name.
+func GenerateNatGatewayIPName(clusterName, subnetName string) string {
+	return fmt.Sprintf("pip-%s-%s-natgw", clusterName, subnetName)
```

Comment:
Not sure if two different subnets in two different clusters can have the same name, so to be sure I'm using the cluster name.

Reply:
the subnet name should already have the clusterName as part of its name by default, and if it doesn't (eg. custom name) two subnets with the same name can't be in the same RG so I think it's okay to remove clusterName from here.

Reply:
@fiunchinho did you want to update this?